File: chap4/ParValeur.ml
(* Ces programmes sont sous licence CeCILL-B V1. *)

let i = ref 0

let h j =
  let j = j + 1 in
  print_int !i; print_newline ();
  print_int j; print_newline ();
  let k = j + !i in
  i := 5;
  k

let _ =
  let m = 1 in
      i := 10;
  print_int m; print_newline ();
  let n = h m in
  print_int m; print_newline ();
  print_int n; print_newline ();
  print_int !i; print_newline ()

File: chap4/ParValeur.java
// Ces programmes sont sous licence CeCILL-B V1.
class ParValeur {
  static int i;

  static int h (int j) { 
    int k;
    j = j + 1;
    System.out.println(i);
    System.out.println(j);
    k = j + i;
    i = 5;
    return k;}

  public static void main (String [] args) {
    int m, n;
    m = 1;
    i = 10;
    System.out.println(m);
    n = h(m);
    System.out.println(m);
    System.out.println(n);
    System.out.println(i);}}
